Tasks/Common/azure-arm-rest/azure-arm-endpoint.ts

```diff
@@ -0,0 +1,118 @@
+import tl = require('vsts-task-lib/task');
+import Q = require('q');
+import webClient = require("./webClient");
+import { AzureEndpoint } from "./azureModels";
+export class AzureRMEndpoint {
+    public _endpoint: AzureEndpoint;
+    private _connectedServiceName: string;
+    // Add an entry here and separate function for each new environment
+    private _environments = {
+        'AzureStack': 'azurestack'
+    }
+
+    constructor(connectedServiceName: string) {
+        this._connectedServiceName = connectedServiceName;
+        this._endpoint = null;
+    }
+
+    public getEndpoint() {
+        let dataDeferred = Q.defer<AzureEndpoint>();
+        if(!!this._endpoint) {
+            dataDeferred.resolve(this._endpoint);
+        }
+        else {
+            this._endpoint = {
+                subscriptionID: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionid', true),
+                subscriptionName: tl.getEndpointDataParameter(this._connectedServiceName, 'subscriptionname', true),
+                servicePrincipalClientID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalid', false),
+                servicePrincipalKey: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'serviceprincipalkey', false),
+                environmentAuthorityUrl: tl.getEndpointDataParameter(this._connectedServiceName, 'environmentAuthorityUrl', true),
+                tenantID: tl.getEndpointAuthorizationParameter(this._connectedServiceName, 'tenantid', false),
+                url: tl.getEndpointUrl(this._connectedServiceName, true),
+                environment: tl.getEndpointDataParameter(this._connectedServiceName, 'environment', true),
+                activeDirectoryResourceID: tl.getEndpointDataParameter(this._connectedServiceName, 'activeDirectoryServiceEndpointResourceId', true)
+            } as AzureEndpoint;
+
+            // Initialize Azure Endpoint for specific environment
+            switch(this._endpoint.environment.toLowerCase()) {
+                case this._environments.AzureStack: {
+                    this._getAzureStackData(this._endpoint).then((endpoint) => {
+                        dataDeferred.resolve(endpoint);
+                    }, (error) => {
+                        dataDeferred.reject(error);
+                    })
+                    break;
+                }
+                default: {
+                    this._endpoint.environmentAuthorityUrl = (!!this._endpoint.environmentAuthorityUrl) ? this._endpoint.environmentAuthorityUrl : "https://login.windows.net/";
+                    this._endpoint.activeDirectoryResourceID = this._endpoint.url;
+                    dataDeferred.resolve(this._endpoint);
+                }
+            }
+        }
+
+        return dataDeferred.promise;
+    
+    }
+
+    private _getAzureStackData(endpoint: AzureEndpoint) {
```

Comment:
By defining this method I am assuming you are removing dependency from azurestack-common. Please make sure all modules that are dependent on azurestack-common will get updated. Let me know if you need help regarding azure stack validations.

Reply:
the dependency is removed only for Azue App Service Manage task, and will be removing for Azure App Service Deploy task too. 
As discussed, it may be a code redundancy, will inform the other task owners as well. (Azure Key vault & Azure RG task)